nixos/modules/services/misc/nixos-manual.nix

```diff
@@ -58,6 +58,15 @@ let
       exec "$browser" ${entry}
     '';
 
+  desktopItem = pkgs.makeDesktopItem {
+    name = "nixos-manual";
+    desktopName = "NixOS Manual";
+    genericName = "View NixOS documentation in a web browser";
+    # TODO: find a better icon (Nix logo + help overlay?)
+    icon = "system-help";
+    exec = "xdg-open ${entry}";
```

Comment:
I guess it should be if at least one desktop env is installed system-wide.
On a server with no xserver the `xdg-open` command is not available. But since you just made the creation of the desktop item conditional it should be fine.